engine/baml-lib/parser-database/src/walkers/arg.rs

```diff
@@ -0,0 +1,74 @@
+use super::{ClassWalker, ClientWalker, ConfigurationWalker, EnumWalker, Walker};
+use crate::{
+    ast::{self, WithName},
+    printer::{serialize_with_printer, WithSerializeableContent},
+    types::FunctionType,
+    ParserDatabase, WithSerialize,
+};
+use either::Either;
+use internal_baml_schema_ast::ast::{ArgumentId, Identifier, ValExpId, WithIdentifier, WithSpan};
+
+pub type ArgWalker<'db> = super::Walker<'db, (ValExpId, bool, ArgumentId)>;
+
+impl<'db> ArgWalker<'db> {
+    /// The ID of the function in the db
+    pub fn function_id(self) -> ast::ValExpId {
+        self.id.0
+    }
+
+    /// The AST node.
+    pub fn ast_function(self) -> &'db ast::ValueExprBlock {
+        &self.db.ast[self.id.0]
+    }
+
+    /// The AST node.
+    pub fn ast_arg(self) -> (Option<&'db Identifier>, &'db ast::BlockArg) {
+        match self.id.1 {
+            true => {
+                let args = self.ast_function().input();
+                let res = &args.expect("Expected input args")[self.id.2];
+                (Some(&res.0), &res.1)
+            }
+
+            false => {
+                let output = self.ast_function().output();
+                let res = output.expect("Error: Output is undefined for function ID");
```

Comment:
ssame thing

Reply:
I'm pretty sure this expect is always safe, because if the argument exists, it will be of input or output and be matched directly. If it doesn't exist, then the arg would not have been created in the first place. Prior to this PR, we had a function specific ast node so input/output args were required, but now we use that same ast node structure for other structs as well which have these as optional. However, since this code is specifically for functions, it is safe. Can discuss tomorrow.